### PR TITLE
Add topic auto-creation.

### PR DESCRIPTION
**Pull request by kbourgoin**
Adds topic auto-creation to the Python implementation for #128.  I went with a very simple implementation with hard-coded retry and backoff values.  I'm not sure if we need to have these configurable, but could probably repurpose some of our settings if so.

### CODE REVIEW

File: pykafka/pykafka/cluster.py [hunk 1]
```diff
@@ -3,16 +3,38 @@
...
+    """Dictionary which will attempt to auto-create unknown topics."""
+
+    def __init__(self, cluster, *args, **kwargs):
+        super(TopicDict, self).__init__(*args, **kwargs)
+        self._cluster = weakref.proxy(cluster)
+
+    def __missing__(self, key):
+        logger.info('Topic %s not found. Attempting to auto-create.', key)
+        # Auto-creating will take a moment, so we try 3 times.
+        for i in xrange(3):
```
Comment (emmett9001) [inline pykafka/pykafka/cluster.py]: There are about 15 different places where we do backoff logic like this. I'm wondering if there's a way we can standardize.

File: pykafka/pykafka/cluster.py [hunk 2]
```diff
@@ -3,16 +3,38 @@
...
+            self._cluster.brokers[0].request_metadata(topics=[key])
```
Comment (emmett9001) [inline pykafka/pykafka/cluster.py]: I'm not sure what this does in the context of creating a new topic, but it seems important. Maybe it should be rolled up into a convenience method along with the next line.

**Comment by emmett9001** [comment]
Nice and simple. Looks good!

File: pykafka/pykafka/cluster.py (hunk 1 above)
Comment (kbourgoin) [inline pykafka/pykafka/cluster.py]: I agree, but I'm going to pass for the sake of this PR.  We could definitely use some standardization there.

File: pykafka/pykafka/cluster.py (hunk 2 above)
Comment (kbourgoin) [inline pykafka/pykafka/cluster.py]: Yeah, that's a good point. I'll add it as a private method and have a docstring explaining why it does what it does.

**Pull request merged**